jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -238,23 +261,32 @@ private void generateSchema(Class<?> schemaClass) throws MojoExecutionException
     private List<PotentialSchemaClass> getAllClassNames() {
         if (this.allTypes == null) {
             Scanner subTypeScanner = Scanners.SubTypes.filterResultsBy(c -> true);
-            URLClassLoader urlClassLoader = this.getClassLoader();
             ConfigurationBuilder configBuilder = new ConfigurationBuilder()
-                    .forPackage("", urlClassLoader)
                     .addScanners(subTypeScanner);
-            if (urlClassLoader != null) {
-                configBuilder.addUrls(urlClassLoader.getURLs());
+            if (annotations != null && !annotations.isEmpty()) {
+                configBuilder.addScanners(Scanners.TypesAnnotated);
             }
+
+            configBuilder.addUrls(getUrls(classpath));
+
             Reflections reflections = new Reflections(configBuilder);
-            Stream<PotentialSchemaClass> allTypesStream = reflections.getAll(subTypeScanner)
+            Set<String> allTypesSet;
+            if (annotations != null && !annotations.isEmpty()) {
+                allTypesSet = reflections.get(Scanners.TypesAnnotated.with(annotations));
+            } else {
+                allTypesSet = reflections.getAll(subTypeScanner);
+            }
```

Comment:
**suggestion (not-blocking)**: I personally prefer avoiding double negation (`if (not x) a else b`) and rather swap the `if`/`else` blocks in such a case (`if (x) b else a`).
Makes the code slightly more readable in my view.
```suggestion
            if (annotations == null || annotations.isEmpty()) {
                allTypesSet = reflections.getAll(subTypeScanner);
            } else {
                allTypesSet = reflections.get(Scanners.TypesAnnotated.with(annotations));
            }
```

Reply:
Changed that